WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift

```diff
@@ -497,6 +498,7 @@ class GutenbergViewController: UIViewController, PostEditor, FeaturedImageDelega
     }
 
     func requestHTML(for reason: RequestHTMLReason) {
+        wpAssert(requestHTMLReason == nil, "HTML request is in progress")
```

Comment:
I keep hitting this assertion in the second scenario with a long post (5000 words). Two requests get called in a row before the first one completes. One for autoSave and another for custom callback. 

If I test the same case without an assertion, then `requestHTML` gets called twice, but `gutenbergDidProvideHTML` gets only called once. If it finished with reason `autosave`, then the callback() never gets called, and the more menu gets stuck in `loading..`  state.

<img width="281" alt="image" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/e31e5853-7715-41ab-88ca-c984fdd3fd41">

Reply:
Good catch. I added a bit of Combine to resolve it. I'm not sure it's completely sound, but it should work for this use case.

Reply:
Yes, it does do the trick.